jupyter_collaboration/handlers.py

```diff
@@ -297,6 +301,13 @@ def _emit(self, level: LogLevel, action: str | None = None, msg: str | None = No
 
         self.event_logger.emit(schema_id=JUPYTER_COLLABORATION_EVENTS_URI, data=data)
 
+    def _emit_awareness_event(self, user: str, action: str, msg: str | None = None) -> None:
```

Comment:
(no action needed) I think we should prefer declaring the type of `msg` as `Optional[str]` instead of `str | None`, since the vertical bar syntax for type unions is [only available for Python 3.10+](https://peps.python.org/pep-0604/). I thought the convention in Jupyter packages was to support every currently maintained version of Python, [the lowest being Python 3.8 at present](https://devguide.python.org/versions/).

However, I also noticed that the rest of the source code uses the `str | None`. I've [opened an issue](https://github.com/jupyterlab/jupyter-collaboration/issues/248) so we can discuss this as a team. No action needed on this PR, just a comment.

Reply:
It's fine to use `str | None`. Jupyter collaboration [supports Python 3.8 and above](https://github.com/jupyterlab/jupyter-collaboration/blob/8fee21f7cd338057c69febb5508f9b6c8bfd2392/pyproject.toml#L12). Union types as `X | Y` are allowed for python<3.10 with:
```py
from __future__ import annotations
```